### PR TITLE
Feature: Integrate firebase

### PR DESCRIPTION
**Pull request by Duc-Lam-jh**

### CODE REVIEW

File: src/app/redux/auth/authActions.js [hunk 1]
```diff
@@ -30,36 +34,42 @@ const setErrorWrongPassword = () => {
...
+            localStorage.setItem("userId", userData.id);
+            localStorage.setItem("userName", userData.name);
+          })
+          .catch(error => {
+            dispatch(setError(error.message));
+          })
+      })
+      .catch(error => {
+        dispatch(setError(error.message));
+      })
```
Comment (anh-dinh-jh) [inline src/app/redux/auth/authActions.js]: Is there a way to make this one cleaner?

File: src/app/data/devicesActions.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  return devices;
+}
+
+const getDeviceById = async id => {
+  const snap = await getDoc(doc(devicesCollection, id));
+  if (snap.exists()) {
+    return { ...snap.data(), id: id };
+  }
+}
```
Comment (anh-dinh-jh) [inline src/app/data/devicesActions.js]: What if `snap.exists()` is false?

File: src/app/data/devicesActions.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const response = await getDocs(devicesCollection);
+  const documents = response.docs;
+
+  const devices = [];
+  documents.forEach(item => {
+    devices.push({
+      id: item.id,
+      ...item.data()
+    })
+  })
```
Comment (anh-dinh-jh) [inline src/app/data/devicesActions.js]: Could we use `map` for this one? If so, please find all similar cases and fix them

File: src/app/data/devicesActions.js (hunk 2 above)
Comment (Duc-Lam-jh) [reply src/app/data/devicesActions.js]: My idea was if the snap doesn't exist, the function will return undefined (so I leave that case to be blank). But now as I see it creates confusion, I decided to change it to return null.

File: src/app/redux/auth/authActions.js (hunk 1 above)
Comment (Duc-Lam-jh) [reply src/app/redux/auth/authActions.js]: I shortened it by making the saving data to local storage part a separate function

File: src/app/redux/auth/authActions.js [hunk 4]
```diff
@@ -30,36 +35,41 @@ const setErrorWrongPassword = () => {
...
+          .then(userData => {
+            if(userData !== null){
+              saveUserToBrowser({...userData});
+              dispatch(setActiveUser({ ...userData }));
+            }
+            else{
+              dispatch(setErrorUserNotExist());
+            }
+          })
+      })
```
Comment (anh-dinh-jh) [inline src/app/redux/auth/authActions.js]: Is there any solution to avoid the loop of `then`?

Comment (Duc-Lam-jh) [reply src/app/redux/auth/authActions.js]: we can turn the returned function into an async function and use await instead of .then

**Pull request merged**